tests/gold_tests/autest-site/when.test.ext

```diff
@@ -19,18 +19,66 @@ When extensions.
 
 from autest.api import AddWhenFunction
 import hosts.output as host
+import os
+import re
 
 
-def FileContains(haystack, needle):
+def FileContains(haystack, needle, desired_count=1):
```

Comment:
Does Au Test allow this to be a functor?  Seems like it should pre-compile the regular expression, and keep track of how much of the file it has already scanned.  But perhaps beyond the scope of this PR.

Reply:
That's a good point. I should look into pre-compiling the regex.